tools/ReleaseTool/ReleaseCommand.cs

```diff
@@ -0,0 +1,203 @@
+                    }
+                }
+                
+                // Create release
+                CreateRelease(gitHubClient, gitHubRepo, gitClient);
+
+                Console.WriteLine("");
+                Console.WriteLine(string.Format("Release Complete! Release hash: {0}", gitClient.GetHeadCommit().Sha));
+            }
+            catch (Exception e)
+            {
+                Console.Error.WriteLine("ERROR: Unable to release candidate branch. Error: {0}", e.ToString());
+                return 1;
+            }
+            finally
+            {
+                gitClient.Dispose();
+            }
+
+            return 0;
+        }
+
+        private string BranchName()
+        {
+            return CommandsCommon.ReleaseBranchName(options.Version);
+        }
+
+        private void UpdateGdkVersion(GitClient gitClient)
+        {
+            if (!ShouldUpdateGdkVersion())
+            {
+                return;
+            }
+
+            CommandsCommon.UpdateGdkVersion(gitClient, options.GdkVersion);
+
+            gitClient.Commit(string.Format(UpdateGdkCommitMessageTemplate, options.Version));
+        }
+
+        private bool ShouldUpdateGdkVersion()
+        {
+            return !string.IsNullOrEmpty(options.GdkVersion);
+        }
+
+        private void CreateRelease(GitHubClient gitHubClient, Octokit.Repository gitHubRepo, GitClient gitClient)
+        {
+            Console.WriteLine("Running packer...");
+            var package = Packer.Program.Package(Environment.CurrentDirectory);
+
+            var commitish = gitClient.GetHeadCommit().Sha;
+            
+            var releaseBody = GetReleaseNotesFromChangeLog();
+            var release = gitHubClient.CreateDraftRelease(gitHubRepo, options.Version, releaseBody, options.Version,
+                commitish);
+
+
+            using (var reader = File.OpenRead(package))
+            {
+                gitHubClient.AddAssetToRelease(release, Path.GetFileName(package), PackageContentType, reader);
+            }
+        }
+
+        private static string GetReleaseNotesFromChangeLog()
+        {
+            if (!File.Exists(ChangeLogFilename))
+            {
+                Console.WriteLine("Could not get draft release notes, as the change log file, {0}, does not exist.", ChangeLogFilename);
+                return string.Empty;
+            }
+
+            Console.WriteLine("Reading {0}...", ChangeLogFilename);
+
+            var releaseBody = new StringBuilder();
+            var changedSection = 0;
+
+            using (var reader = new StreamReader(ChangeLogFilename))
+            {
+                while (!reader.EndOfStream)
+                {
+                    var line = reader.ReadLine();
+                    
+                    if (line.StartsWith("## "))
+                    {
+                        changedSection += 1;
+
+                        if (changedSection == 3)
```

Comment:
I think using a Markdown library would help make these sections more legible

Reply:
(See comment above, the AST in markdig / other markdown libraries; is quite difficult to manipulate). For this particular use case it might easier, but its not possible for writing Markdown :(